ldk/javascript/src/search/mapper.ts

```diff
@@ -0,0 +1,100 @@
+import { Config, Index, Document, Field, FieldType, SearchResult } from './index';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const handleCaughtError = (reject: (reason?: any) => void, error: Error, type: string): void => {
+  console.error(`Received error calling ${type}: ${error.message}`);
+  reject(error);
+};
+
+export const mapToFieldType = (type: FieldType | undefined): OliveHelps.FieldType =>
+  type as OliveHelps.FieldType;
+export const mapToField = (field: Field): OliveHelps.Field => ({
+  name: field.name,
+  displayName: field.displayName,
+  type: mapToFieldType(field.type),
+});
+
+export const mapToFields = (fields: Field[]): OliveHelps.Field[] =>
+  fields.map((f) => mapToField(f));
+
+export const mapToSearchResult = (searchResult: OliveHelps.SearchResult): SearchResult => ({
+  ...searchResult,
+});
+
+export const mapToDocument = (document: Document): OliveHelps.Document => ({
+  name: document.name,
+  data: document.data,
+  fields: document.fields ? mapToFields(document.fields) : undefined,
+});
+
+export const mapToDocuments = (documents: Document[]): OliveHelps.Document[] => {
+  const docs: OliveHelps.Document[] = [];
+  documents.forEach((d) => docs.push(mapToDocument(d)));
+  return docs;
+};
+
+export const mapToConfig = (config: Config): OliveHelps.Config => ({
+  ...config,
+});
+
+export const mapToIndex = (index: OliveHelps.Index): Index => ({
+  search: (term: string): Promise<SearchResult> =>
+    new Promise<SearchResult>((resolve, reject) => {
+      try {
+        index.search(term, (error: Error | undefined, searchResult: SearchResult) => {
+          if (error) {
+            console.error(`Received error on result: ${error.message}`);
```

Comment:
It would make sense to handle error in here the same way as you did in the catch block. Since you already abstracted the handler:
`handleCaughtError(reject, error as Error, 'search');`
Same goes for the rest of the methods.